Introduce Deployment freeze (#1190)
OctopusDeploy/blog: blog/2024-12/deployment-freeze/index.md

---

> title: Introducing Deployment Freezes

Reviewer: I think we need a different title given that deployment freezes already exist with the product. Maybe something similar to below?
More control over Deployment Freezes: Recurring freezes and freeze by tenant

description: Octopus Deploy introduces a new feature called deployment freezes, allowing you to restrict deployments during specific periods to ensure system stability and meet business requirements.
author: huy.nguyen@octopus.com
visibility: public
published: 2024-12-19-1400
metaImage: blogimage-deploymentfreezes-2024-750x400.png
bannerImage: blogimage-deploymentfreezes-2024-750x400.png
bannerImageAlt: Two people looking at an Octopus Deploy interface showing deployment versions with a winter theme and snowflakes
tags: 
- Product
- Deployment
---

Have you ever needed to stop deployments during busy periods? Maybe during holidays when most of your team is away, or during big sales events? We've got great news! We're launching deployment freezes in Octopus Deploy to help you manage these situations better.

## Why Do You Need Deployment Freezes?

Let's face it - there are times when deploying new changes is just too risky. Think about:
- Holiday seasons when your team is short-staffed
- Critical business periods like tax season or elections

> - Important demos or testing cycles when you need your systems stable

Reviewer: A couple of scenarios to add;

- You need to prevent deployments to production during business hours
- Your customers / branches are modelled with tenants and require more granular control around when deployments to those customers can proceed

Our new feature helps you handle all these cases. It lets you control exactly when and where deployments can happen, keeping your systems stable when it matters most.

## Setting Up Your Freezes

Setting up a deployment freeze is straightforward. You can create them for right now or schedule them for later. You get to choose:
- A clear name so everyone knows what the freeze is for
- When it starts and ends
- How long it lasts
- Which projects, environments, and tenants it affects

> Need to set up regular freezes for maintenance? No problem! You can set up automated recurring freezes that match your schedule.

Reviewer: Need to set up regular maintenance windows? No problem! You can set up automated recurring freezes that match your schedule.

![Deployment Freeze Detail](deployment-freeze-detail.png "width=500")

## Choosing What to Freeze

> You get two main ways to control your freezes: project scope and tenant scope.

Reviewer 2: ```suggestion
You can control your freezes by scoping them to projects or tenants.
```

> With project scope, you can freeze specific projects in certain environments. For example, you might want to freeze your online store deployments during Black Friday but keep working on your internal tools.

Reviewer 2: ```suggestion
With project scopes, you can freeze by environment. For example, you might want to freeze your online store production deployments during Black Friday but keep working on your development and test environments.
```

> For regular maintenance windows or recurring business cycles, you can set up automated recurring freezes that align with your organization's needs.

Reviewer 3: Each line proceeding an image needs a blank line before the image or it ends up inlining the paragraph and image.

> ![Deployment Freeze Detail](deployment-freeze-detail.png)

Reviewer 3: All of the images need to have ` "width=500"` added after the filename. I think this keeps the images down to a reasonable size when rendered.

![Project Scope](deployment-freeze-project-scope.png "width=500")

> Tenant scope lets you get even more specific. You can freeze deployments for particular tenants in selected projects and environments. This is super helpful if you're managing systems across different time zones or regions.

Reviewer 2: ```suggestion
Tenant scope lets you get even more specific. You can freeze deployments for particular tenants in selected projects and environments. This is helpful if you're managing systems across different time zones or regions.
```

> Deployment freezes in Octopus Deploy offer flexible scoping options for precise control over deployment restrictions. The scoping system provides two powerful options to give you granular control over your deployment activities.

Reviewer 3: ```suggestion
Deployment freezes in Octopus Deploy offer flexible scoping options for precise control over deployment restrictions. The scoping configuration provides two powerful options to give you granular control over your deployment activities.
```

## Setting Up Regular Freezes

Need to set up regular freeze periods? Our recurring schedule feature has got you covered. You can:
- Set up different freezes for each global region
- Match your maintenance windows
- Create daily protection windows

The system lets you set up these recurring patterns:
- Daily windows to protect business hours
- Weekly schedules for regular maintenance
- Monthly patterns for release cycles
- Annual schedules for recurring business events

![Deployment Freeze Recurrence](deployment-freeze-recurrence.png "width=500")

## Override Capabilities and Audit Trail

We know that sometimes you need to make exceptions. That's why we've made it easy for authorized users to override active freezes. They just need to provide a reason, which we'll keep track of in the audit trail.

The audit trail keeps a record of:
- Freeze creation and modification
- Deployment Freeze Override and their justifications
- Deployment attempts during freeze periods

![Screenshot of the Deployment Freeze interface](deployment-freeze-override.png "width=500")

## Automation and Integration

> You can automate and integrate deployment freezes into your workflows. Use the Octopus REST API and Terraform provider. For example, you can use the Go client to automate the creation and management of deployment freezes as part of your CI/CD pipeline. The Terraform provider includes an `octopusdeploy_deployment_freeze` resource. It can automate freeze configuration. For more information, see the following resources:

Reviewer 2: ```suggestion
You can automate and integrate deployment freezes into your workflows using the Octopus REST API or the Terraform provider. For example, you can use the Go client to automate the creation and management of deployment freezes as part of your CI/CD pipeline. The Terraform provider offers an `octopusdeploy_deployment_freeze` resource for creating and managing deployment freezes. For more information, see the following resources:
```

Tenant scope lets you get even more specific. You can freeze deployments for particular tenants in selected projects and environments. This is super helpful if you're managing systems across different time zones or regions.

![Tenant Scope](deployment-freeze-tenant-scope.png "width=500")

By mixing these options, you can create freezes that match exactly what your team needs.

## What Happens During a Freeze?

Here's what you can expect when a freeze is active:
- Existing deployments that started before the freeze will continue to completion
- New deployments attempted during the freeze will need to provide an override reason to be able to deploy
- Scheduled deployments that would start during a freeze period will not execute
- Automatic deployments based on deployment target triggers are allowed to ensure your deployment targets stay updated when scaling up
- Other automatic deployments, such as scheduled deployments or automatic lifecycle promotions, are blocked during the freeze

## Setting Up Regular Freezes

Need to set up regular freeze periods? Our recurring schedule feature has got you covered. You can:
- Set up different freezes for each global region

> - Match your maintenance windows

Reviewer 3: The problem with describing it as a maintenance window is that is the opposite of a freeze, for example, octopus cloud has a 2 hour maintenance window per day, that means that the freeze is the other 22 hours, per day.
A better example might be something like the end of month or end of financial year processing where they can't deploy during job runs.

- Create daily protection windows

The system lets you set up these recurring patterns:
- Daily windows to protect business hours
- Weekly schedules for regular maintenance

> - Monthly patterns for release cycles

Reviewer 3: This example is the same as I described above, where it feels like a release window rather than a release freeze.
